core/test/qa_Messages.cpp

```diff
@@ -18,509 +20,633 @@ auto boost::ut::cfg<boost::ut::override> = boost::ut::runner<boost::ut::reporter
+
+                expect(eq(fromBlock.streamReader().available(), 1UZ)) << "didn't receive reply message";
+                const Message reply = returnReplyMsg(fromBlock);
+                expect(reply.cmd == Final) << fmt::format("mismatch between reply.cmd = {} and expected {} command", reply.cmd, Final);
+                expect(eq(reply.serviceName, unitTestBlock.unique_name));
+                expect(eq(reply.clientRequestID, "client#42"s));
+                expect(eq(reply.endpoint, std::string(block::property::kEcho)));
+                expect(!reply.data.has_value());
+            };
+        };
+
+        "Block<T>-level lifecycle::State tests"_test = [] {
+            gr::MsgPortOut toBlock;
+            TestBlock<int> unitTestBlock({ { "name", "UnitTestBlock" } });
+            gr::MsgPortIn  fromBlock;
+
+            expect(eq(ConnectionResult::SUCCESS, toBlock.connect(unitTestBlock.msgIn)));
+            expect(eq(ConnectionResult::SUCCESS, unitTestBlock.msgOut.connect(fromBlock)));
+
+            "get - state"_test = [&] {
+                sendMessage<Get>(toBlock, "" /* serviceName */, block::property::kLifeCycleState /* endpoint */, {} /* data  */, "client#42");
+                expect(nothrow([&] { unitTestBlock.processScheduledMessages(); })) << "manually execute processing of messages";
+
+                expect(eq(fromBlock.streamReader().available(), 1UZ)) << "didn't receive reply message";
+                const Message reply = returnReplyMsg(fromBlock);
+                expect(reply.cmd == Final) << fmt::format("mismatch between reply.cmd = {} and expected {} command", reply.cmd, Final);
+                expect(eq(reply.serviceName, unitTestBlock.unique_name));
+                expect(eq(reply.clientRequestID, "client#42"s));
+                expect(eq(reply.endpoint, std::string(block::property::kLifeCycleState)));
+                expect(reply.data.has_value());
+            };
+
+            "set - state"_test = [&] {
+                sendMessage<Set>(toBlock, "" /* serviceName */, block::property::kLifeCycleState /* endpoint */, { { "state", "INITIALISED" } } /* data  */);
+                expect(nothrow([&] { unitTestBlock.processScheduledMessages(); })) << "manually execute processing of messages";
+
+                expect(eq(fromBlock.streamReader().available(), 0UZ)) << "should not receive a reply";
+                expect(unitTestBlock.state() == lifecycle::State::INITIALISED);
+            };
+
+            "set - state - error cases"_test = [&] {
+                sendMessage<Set>(toBlock, "" /* serviceName */, block::property::kLifeCycleState /* endpoint */, {} /* no data  */);
+                expect(nothrow([&] { unitTestBlock.processScheduledMessages(); })) << "manually execute processing of messages";
+                expect(eq(fromBlock.streamReader().available(), 1UZ)) << "should not receive a reply";
```

Comment:
There is no reply but error is still sent. Change to "should have one error".

Reply:
done.